mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -556,6 +571,7 @@ class Word2VecModel private[spark] (
       .sortBy(- _._2)
       .take(num + 1)
       .tail
+      .map(v => (if (vecNorm == 0) v else (v._1, v._2 / vecNorm)))
```

Comment:
@ygcao please change this line to `.map { case (word, cosVec) => (word, if (vecNorm == 0.0) 0.0 else (cosVec / vecNorm)) }` as per discussion above.